itemlist.lua

```diff
@@ -0,0 +1,231 @@
+	end
+	minetest.log('warning',	'['..modname..'] get_player_name could not identify player.')
+end
+
+minetest.register_on_leaveplayer(function(player)
+	local playername = get_player_name(player)
+	if playername then contexts[playername] = nil end
+end)
+
+local function new_context(player, context)
+	local playername = get_player_name(player)
+	if playername then
+		contexts[playername] = context
+		contexts[playername].playername = playername
+		return contexts[playername]
+    end
+end
+
+local function get_context(player)
+	local playername = get_player_name(player)
+	if playername then
+		if contexts[playername] then
+			return contexts[playername]
+		else
+			minetest.log('warning', '['..modname..'] Context not found for player "'..playername..'"')
+		end
+	end
+end
+
+local function update_context(player, changes)
+	local playername = get_player_name(player)
+	if playername then
+		if not contexts[playername] then
+			contexts[playername] = { playername = playername }
+		end
+		for key, value in pairs(changes) do
+			contexts[playername][key] = value
+		end
+	end
+end
+
+-- Show node formspec functions
+
+local function show_node_formspec(player, pos)
+	local meta = minetest.get_meta(pos)
+	local playername = get_player_name(player)
+
+	-- Decontextualize formspec
+	local fs = meta:get_string('formspec')
+
+	-- Change context and currrent_name references to nodemeta references
+	fs = fs:gsub("current_name", "nodemeta:"..pos.x..","..pos.y..","..pos.z)
+	fs = fs:gsub("context", "nodemeta:"..pos.x..","..pos.y..","..pos.z)
+
+	-- Change all ${} to their corresponding metadata values
+	local s, e
+	repeat
+		s, e = fs:find('%${.*}')
+		if s and e then
+			fs = fs:sub(1, s-1)..
+				minetest.formspec_escape(meta:get_string(fs:sub(s+2,e-1)))..
+				fs:sub(e+1)
+		end
+	until s == nil
+
+	-- Find node on_receive_fields
+	local ndef = minetest.registered_nodes[minetest.get_node(pos).name]
+
+	if ndef and ndef.on_receive_fields then
+		update_context(player, { on_receive_fields = ndef.on_receive_fields } )
+	end
+	update_context(player, { node_pos = pos } )
+
+	-- Show formspec
+	minetest.show_formspec(playername, modname..':context_formspec', fs)
+end
+
+minetest.register_on_player_receive_fields(function(player, formname, fields)
+	if formname == modname..':context_formspec' then
+		local context = get_context(player)
+		if context == nil then return end
+
+		if context.on_receive_fields then
+			context.on_receive_fields(context.node_pos, '', fields, player)
+		end
+	end
+end)
+
+-- Specific functions
+
+local function get_nb_of_pages(context)
+	return math.ceil((#context.items + 1) / (style.lines * style.cols))
+end
+
+local function item_list_prepare(item_list)
+	local list = {}
+	for _, name in ipairs(item_list) do
+		local ndef = minetest.registered_items[name]
+		if ndef then
+			list[#list+1] = ndef
+		end
+	end
+	table.sort(list, function(a,b) return a.description < b.description end)
+	return list
+end
+
+local function show_fs(player)
```

Comment:
More descriptive name: `show_item_list_formspec`

Reply:
Done in 1c132d84b0bd4a862c6d29f73a83d6c1ee372259